src/v/cloud_storage/partition_manifest.cc

```diff
@@ -201,7 +201,10 @@ partition_manifest::lw_segment_meta::convert(const segment_meta& m) {
       .segment_term = m.segment_term,
       .size_bytes = m.sname_format == segment_name_format::v1 ? 0
                                                               : m.size_bytes,
-      .sname_format = m.sname_format};
+      .sname_format = m.sname_format,
+      .may_have_tx_index = m.sname_format == segment_name_format::v3
+                           && m.metadata_size_hint != 0,
```

Comment:
Not sure I understand why we need this. `sname_format` and `metadata_size_hint` are already present in `segment_meta`. `garbage_collect_archive` already uses them to figure out if there's a transactional manifest to delete.